### DISCUSSION TITLE
How to stop function run if serverless function timed out?

### DISCUSSION THREAD

**Discussion by benthecoder** [vercel/community#1788] [Help] [answered]
### Summary

I have a pyTelegramBotAPI bot deployed to Vercel serverless function that calls the ChatGPT API. 

The response often takes >10s, and I saw in the logs it's repeatedly trying to call the function, which can hurt my credits. Is there a way to quit running the function in my code so that Vercel isn't repeatedly trying to call my function?

Also, when I redeploy the app, it still keeps running the previous POST requests. Is there a way to tell Vercel to stop running it and kill the process?

### Additional information

Here's the code that calls GPT

```
@bot.message_handler(commands=["t"])
def chat(message):
    """Generate a response to a user-provided message"""
    response = gpt(message.text)
    bot.reply_to(message, response)
```

My github repo: https://github.com/benthecoder/GPTJournal/tree/main

### Example

https://github.com/benthecoder/GPTJournal

**Comment by steven-tey** [comment, marked as answer]
Hey Ben! Have you checked out @Nutlope's [article on how you can stream data using Edge Functions](https://vercel.com/blog/gpt-3-app-next-js-vercel-edge-functions)? We highly recommend using Edge Functions for streaming responses from ChatGPT API since it gets rid of the timeout constraint and it's also cheaper than Serverless Functions.

Here are the docs for it: https://vercel.com/docs/concepts/functions/edge-functions/streaming

**Reply by benthecoder** [reply]
Thanks for the response, @steven-tey. Do you have examples of how to do streaming in Python?
